src/solver/mha/mha_solver_backward.cpp

```diff
@@ -90,8 +90,8 @@ bool MhaBackward::IsApplicable([[maybe_unused]] const ExecutionContext& context,
 
     auto [N, H, S, D] = miopen::tien<4>(descsBwd.kDesc.GetLengths());
 
-    return MIOPEN_USE_GEMM                                            //
-           && !miopen::IsDisabled(ENV(MIOPEN_DEBUG_ATTN_NAIVE_BWD))   //
+#if MIOPEN_USE_ROCBLAS
```

Comment:
It does not contain any rocblas-specific functionality, except the `USE_ROCBLAS_EX3` which sohuld be properly defoned with or without rocblas.
And ciurently, prior to #3030 `MIOPEN_USE_GEMM` is equal to `MIOPEN_USE_ROCBLAS` so it is not applicable and must not be excluded in that way.

Reply:
When `MIOPEN_USE_GEM` (aka `MIOPEN_USE_ROCBLAS`) is 0, the compiler reports a warning that the rest of the expression is not being evaluated. When `-Werror` is additionally being set, compilation breaks.

Reply:
Here is the effect of `MIOPEN_USE_GEMM` equal to `0` when compiling.

```
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:115:17: warning: code will never be executed [-Wunreachable-code]
  115 |            && ((descsBwd.kDesc.GetType() == miopenFloat)              //
      |                 ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:114:15: warning: code will never be executed [-Wunreachable-code]
  114 |            && descsBwd.kDesc.GetType() == descsBwd.dvDesc.GetType()   //
      |               ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:113:15: warning: code will never be executed [-Wunreachable-code]
  113 |            && descsBwd.kDesc.GetType() == descsBwd.dkDesc.GetType()   //
      |               ^~~~~~~~
C:\...\MIOpen\src\solver\mha\mha_solver_backward.cpp:112:15: warning: code will never be executed [-Wunreachable-code]
  112 |            && descsBwd.kDesc.GetType() == descsBwd.dqDesc.GetType()   //
...
```

Reply:
Ahh, but it is intentional. It looks like false-positive for me.
Ok, don't want fighting with the compiler, let's just exclude it by ifdef. But there will be a merge conflict with #3030